starter/source/model/sets/insert_clause_in_set.F

```diff
@@ -524,94 +524,119 @@ SUBROUTINE INSERT_CLAUSE_IN_SET(SET       ,CLAUSE   ,CLAUSE_OPERATOR,
+         SET%ELLIPSE_ID_MADYMO = CLAUSE%ELLIPSE_ID_MADYMO
+         SET%ELLIPSE_N = CLAUSE%ELLIPSE_N
+         SET%ELLIPSE_XC = CLAUSE%ELLIPSE_XC
+         SET%ELLIPSE_YC = CLAUSE%ELLIPSE_YC
+         SET%ELLIPSE_ZC = CLAUSE%ELLIPSE_ZC
+         SET%ELLIPSE_A = CLAUSE%ELLIPSE_A
+         SET%ELLIPSE_B = CLAUSE%ELLIPSE_B
+         SET%ELLIPSE_C = CLAUSE%ELLIPSE_C
+         CALL MY_ALLOC(SET%ELLIPSE_SKEW,9)
+         SET%ELLIPSE_SKEW(1:9) = CLAUSE%ELLIPSE_SKEW(1:9)
+
+      !ELSEIF ( SET%NB_PLANE > 0 ) THEN
+            ! to be done with RD-14168
+      ELSE
+        !------------------------!
+        !    classic SURFACES    !
+        !------------------------!
+        IF( CLAUSE%NB_SURF_SEG > 0) THEN
+
+          IF( SET%NB_SURF_SEG > 0 ) THEN
+
+              !
+              ! Low level CPP routines / One per operator
+              ! The result is held in CPP Structure 
+              ! until SET is reallocated and unstacked
+              !
+              IF ( CLAUSE_OPERATOR == SET_ADD) THEN                            
+
+                  CALL UNION_SURFACE( SET%SURF_NODES(1,1), SET%SURF_NODES(1,2), 
+     .                                SET%SURF_NODES(1,3), SET%SURF_NODES(1,4),
+     .                                SET%SURF_ELTYP, SET%SURF_ELEM, SET%NB_SURF_SEG,
+     .                                CLAUSE%SURF_NODES(1,1), CLAUSE%SURF_NODES(1,2), 
+     .                                CLAUSE%SURF_NODES(1,3), CLAUSE%SURF_NODES(1,4),
+     .                                CLAUSE%SURF_ELTYP, CLAUSE%SURF_ELEM, CLAUSE%NB_SURF_SEG,
+     .                                NS  )
+
+              ELSEIF ( CLAUSE_OPERATOR == SET_DELETE) THEN
+
+                  CALL DELETE_SURFACE(SET%SURF_NODES(1,1), SET%SURF_NODES(1,2), 
+     .                                SET%SURF_NODES(1,3), SET%SURF_NODES(1,4),
+     .                                SET%SURF_ELTYP, SET%SURF_ELEM, SET%NB_SURF_SEG,
+     .                                CLAUSE%SURF_NODES(1,1), CLAUSE%SURF_NODES(1,2), 
+     .                                CLAUSE%SURF_NODES(1,3), CLAUSE%SURF_NODES(1,4),
+     .                                CLAUSE%SURF_ELTYP, CLAUSE%SURF_ELEM, CLAUSE%NB_SURF_SEG,
+     .                                NS  )
+
+              ELSEIF ( CLAUSE_OPERATOR == SET_INTERSECT)THEN
+
+                  CALL INTERSECT_SURFACE( SET%SURF_NODES(1,1), SET%SURF_NODES(1,2), 
+     .                                    SET%SURF_NODES(1,3), SET%SURF_NODES(1,4),
+     .                                    SET%SURF_ELTYP, SET%SURF_ELEM, SET%NB_SURF_SEG,
+     .                                    CLAUSE%SURF_NODES(1,1), CLAUSE%SURF_NODES(1,2), 
+     .                                    CLAUSE%SURF_NODES(1,3), CLAUSE%SURF_NODES(1,4),
+     .                                    CLAUSE%SURF_ELTYP, CLAUSE%SURF_ELEM, CLAUSE%NB_SURF_SEG,
+     .                                    NS )
+              ELSE
+                  print*,'Unknown clause operator'
```

Comment:
@mircea-altair 

Here a print , I think this should be an error instead.
What do you think about this ?

Sébastien

Reply:
Hi @sebastienVilleneuve ,
The print was there since long. It was not forgot there by my development.
In my opinion, it was left it there by purpose.